packages/core/src/VNodes/AtomicNode.ts

```diff
@@ -0,0 +1,200 @@
+import { VNode, Predicate } from './VNode';
+import { ChildNode } from './ChildNode';
+import { AtomicityError } from '../../../utils/src/errors';
+
+/**
+ * This class provides typing overrides for multiple VNode methods which are
+ * supposed to take parameters but that are unused in the case of atomic nodes.
+ */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+export class AtomicNode extends ChildNode {
+    static readonly atomic: true = true;
```

Comment:
So that TypeScript can warn me if it ever detects that I'm testing if an AtomicNode is not atomic.
Something like that would trigger a warning:
```typescript
const node = new AtomicNode();
if (node.atomic) {
    ...
} else {
    // unreachable
}
```
But of course it's rather hypothetical, because in most cases the node's type will be `VNode` so Typescript won't be able to use the specialized typings I have defined in this file. It might be useful one day if someone makes a mistake at one point though.